.github/workflows/dev_tag_retrieval_pr.yaml

```diff
@@ -0,0 +1,88 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+
+      - name: Fetch current version
+        id: get_version
+        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+
+      - name: Sanitize to remove 'v' prefix
+        id: get_sanitized_version
+        run: echo '##[set-output name=SANITIZED_VERSION;]$(echo ${{ steps.get_version.outputs.VERSION }} | sed 's/v//')'
+
+      - name: Increment Version to dev
+        id: next_version
+        run: |
+          IFS='.' read -ra ver <<< "${{ steps.get_sanitized_version.outputs.SANITIZED_VERSION }}"
+          patch=${ver[2]}
+          minor=${ver[1]}
+          major=${ver[0]}
+          patch=0
+          minor=$((minor+1))
+          echo ::set-output name=NEXT_VERSION::v$major.$minor.$patch-dev
+
+      - name: Update version in file
+        run: sed -i -e "/TKGManagementClusterPluginVersion/ s/= .*/= \"${{ steps.next_version.outputs.NEXT_VERSION }}\"/" pkg/v1/tkg/tkgconfigpaths/zz_bundled_default_bom_files_configdata.go
+
+      - name: Set variables
+        id: vars
+        run: |
+          body=$(cat  << EOF
+          ### What this PR does / why we need it
+          Updating version tag used for BOM retrieval to ${{ steps.next_version.outputs.NEXT_VERSION }}\
+
+          ### Which issue(s) this PR fixes
+
+          Fixes #
+
+          ### Describe testing done for PR
+
+          ### Release note
+          ```release-note
+
+          ```
+
+          ### PR Checklist
+
+          - [x] Squash the commits into one or a small number of logical commits
+          - [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
+          - [x] Ensure PR contains terms all contributors can understand and links all contributors can access
+
+
+          ### Additional information
+
+          #### Special notes for your reviewer
+          EOF
+          )
+          body="${body//'%'/'%25'}"
+          body="${body//$'\n'/'%0A'}"
+          body="${body//$'\r'/'%0D'}"
+          echo ::set-output name=pr_title::"Update the BOM retrieval version to ${{ steps.next_version.outputs.NEXT_VERSION }}"
+          echo ::set-output name=commit_message::"Updating version tag used for BOM retrieval to ${{ steps.next_version.outputs.NEXT_VERSION }}"
+          echo ::set-output name=pr_body::$body
+          echo ::set-output name=pr_branch::"topic/releng/rel${{ steps.next_version.outputs.NEXT_VERSION }}"
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          branch: ${{ steps.vars.outputs.pr_branch }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          base: main
+          title: ${{ steps.vars.outputs.pr_title }}
+          commit-message: ${{ steps.vars.outputs.commit_message }}
+          body: ${{ steps.vars.outputs.pr_body }}
+          delete-branch: true
+          labels: |
+            area/release
```

Comment:
+1 for adding this label

btw, also no new line at the end of the file ;)

Reply:
https://github.com/vmware-tanzu/tanzu-framework/pull/1190#discussion_r751447789